### ISSUE TITLE
Feature request: don't remove files on `--watch`

### ISSUE THREAD

**Issue by DaniGuardiola** [isaacs/tshy#83]
I have a watch script that runs `tshy --watch` and some CSS compilation in parallel. In the output (`dist/esm` and `dist/commonjs`), js files live alongside CSS files. In my normal build script, I simply execute tshy before the CSS compilation, but when in watch mode, tshy detects a filesystem change when a CSS file is written, and then proceeds to remove it. I would like for tshy to leave these files untouched. A reasonable opt-in setting would be okay for me.

**Comment by isaacs** [comment]
Would it be possible to just put the css somewhere else? The problem is that if it doesn't delete the dist folder prior to building, then it's too easy to end up with deleted/renamed files sticking around forever.

**Comment by DaniGuardiola** [comment]
Could it instead just remove js/jsx files (and any empty directories resulting from it? Moving the CSS somewhere else is not a viable option unfortunately.

**Comment by DaniGuardiola** [comment]
I took a look around the code. I think this could be a clean implementation:

- Add an option to [sync-content](https://github.com/isaacs/sync-content) that allows specifying a condition for deleting files or directories, either through a whitelist or blacklist approach.
- In tshy, add an option to the build script that, when syncing tshy-build with dist, only cleans up js/d.ts/etc files (and directories that only contain such files or are empty). This option could be "private", as it probably doesn't make much sense for single runs (user can just do any other dist manipulation afterward).
- Provide an option in --watch along the lines of "clean only js files" or "retain external files" or whatever naming is clearest.

I'd be happy to attempt a contribution. Would you be willing to accept it? Do you have any alternative suggestions?

**Comment by isaacs** [comment]
Yeah, I like the idea of giving sync-content a filter on which files to care about. Maybe a glob pattern or regexp or something.
